src/main/java/com/example/tournamentMaker/randomResultGenerator/RandomResultService.java

```diff
@@ -0,0 +1,186 @@
+import com.example.tournamentMaker.statistics.MatchResult;
+import com.example.tournamentMaker.team.Team;
+import com.example.tournamentMaker.team.TeamRepository;
+import com.example.tournamentMaker.team.player.FootballPlayer;
+import com.example.tournamentMaker.team.player.FootballPosition;
+import com.example.tournamentMaker.tournament.Tournament;
+import com.example.tournamentMaker.tournament.TournamentRepository;
+import com.example.tournamentMaker.tournament.enums.TournamentType;
+import com.example.tournamentMaker.tournament.game.Game;
+import com.example.tournamentMaker.tournament.game.GameRepository;
+import com.example.tournamentMaker.tournament.result.ResultService;
+import com.example.tournamentMaker.tournament.round.CupRoundService;
+import com.example.tournamentMaker.tournament.round.Round;
+import lombok.RequiredArgsConstructor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+@Service
+@RequiredArgsConstructor
+class RandomResultService {
+    private final TournamentRepository tournamentRepository;
+    private final GameRepository gameRepository;
+    private final FootballStatisticsRepository footballStatisticsRepository;
+    private final ResultService resultService;
+    private final CupRoundService cupRoundService;
+    private final TeamRepository teamRepository;
+    private static final int MIN_YELLOW_CARDS_IN_MATCH = 0;
+    private static final int MAX_YELLOW_CARDS_IN_MATCH = 3;
+    private final Logger logger = LoggerFactory.getLogger(RandomResultService.class);
+
+    public void drawLotRoundsResults(RandomResultRequest request) {
+        Tournament tournament = tournamentRepository.findByName(request.getTournamentName())
+                .orElseThrow(() -> {
+                    throw new NoSuchElementException(Constans.NO_TOURNAMENT_FOUND);
+                });
+        String[] rounds = request.getRoundsToDraw().split("-");
+        int firstRound = Integer.parseInt(rounds[0]);
+        int lastRound = Integer.parseInt(rounds[1]);
+
+        drawLotFootballRounds(firstRound, lastRound, tournament);
+    }
+
+    private void drawLotFootballRounds(int firstRound, int lastRound, Tournament tournament) {
+        IntStream.rangeClosed(firstRound - 1, lastRound - 1).forEach(roundNumber -> {
+            try {
+                Round round = tournament.getRounds().get(roundNumber);
+                drawLotFootballRoundResult(round);
+                if (tournament.getTournamentType() == TournamentType.CUP && !isLastRound(tournament, round)) {
+                    cupRoundService.createNextRoundSchedule(tournament, round);
+                }
+            } catch (ArrayIndexOutOfBoundsException e) {
+                logger.error("There is no round with index " + (roundNumber - 1) +
```

Comment:
nie wiem czy sam log error jest wystarczajacy. Bo user nie wie nic. Bo Twoj endpoint i tak zwroci cos 
w kazdy=m razie odda 200 wiec te logi nic nie dadza oprocz tego ze potem w nie moze ktos wejsc jak user powie Panie Kamilu cos sie zjebalo losuje i gowno na froncie nie dostaje zadnego bledu a jak nie bylo nic tak nic nie ma. I to jest kolejne poklosei tego ze wrzucasz voida w 90% aplikacji

Reply:
lepszy boolean czy udało się wylosować czy się nie udało?